bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java

```diff
@@ -661,29 +661,41 @@ private void addAccountsRelationship(long account1_id, long account2_id, Blackbo
 					CommunicationsFilter.DeviceFilter.class.getName(),
 					CommunicationsFilter.RelationshipTypeFilter.class.getName()
 			));
-			String innerQueryfilterSQL = getCommunicationsFilterSQL(filter, applicableInnerQueryFilters);
-
-			String innerQueryTemplate
+			String relationshipFilterSQL = getCommunicationsFilterSQL(filter, applicableInnerQueryFilters);
+			
+			String relationshipLimitSQL = "";
+			for (CommunicationsFilter.SubFilter subFilter : filter.getAndFilters()) {
+				if(subFilter.getClass().getName().equals(CommunicationsFilter.MostRecentFilter.class.getName())) {
+					relationshipLimitSQL =  subFilter.getSQL(this);
```

Comment:
Breaking out of the loop seems like desired behavior here once the relationshipLimitSQL is set